src/mat/4C_mat_inelastic_defgrad_factors.cpp

```diff
@@ -26,13 +26,61 @@
 #include <Teuchos_ParameterList.hpp>
 #include <Teuchos_StandardParameterEntryValidators.hpp>
 
+#include <map>
 #include <memory>
 #include <stdexcept>
 #include <utility>
 
 FOUR_C_NAMESPACE_OPEN
 namespace
 {
+  // enum class for error types in InelasticDefgradTransvIsotropElastViscoplast, used for triggering
+  // the substepping procedures
+  enum class ErrorType
+  {
+    NegativePlasticStrain = 1,  // negative plastic strain which does not allow for evaluations
```

Comment:
This is helpful since I need to "translate" from an int to an ErrorType when communicating with the viscoplastic law. E.g. when returning the error status from `evaluate_plastic_strain_rate` in  `4C_mat_vplast_reform_johnsoncook.cpp`, I want to convert this int to an error type inside InelasticDefgradTransvIsotropElastViscoplast without providing the ErrorType class also to the viscoplasticity law.